pkg/operator/imageconfig.go

```diff
@@ -0,0 +1,251 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	corev1informers "k8s.io/client-go/informers/core/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	configapi "github.com/openshift/api/config/v1"
+	configset "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	routev1informers "github.com/openshift/client-go/route/informers/externalversions/route/v1"
+	routev1lister "github.com/openshift/client-go/route/listers/route/v1"
+
+	"github.com/openshift/cluster-image-registry-operator/pkg/defaults"
+	"github.com/openshift/cluster-image-registry-operator/pkg/resource"
+)
+
+// ImageConfigController controls image.config.openshift.io/cluster.
+//
+// Watches for changes on image registry routes and services, updating
+// the resource status appropriately.
+type ImageConfigController struct {
+	configClient  configset.ConfigV1Interface
+	routeLister   routev1lister.RouteNamespaceLister
+	serviceLister corev1listers.ServiceNamespaceLister
+	cachesToSync  []cache.InformerSynced
+	queue         workqueue.RateLimitingInterface
+}
+
+func NewImageConfigController(
+	configClient configset.ConfigV1Interface,
+	routeInformer routev1informers.RouteInformer,
+	serviceInformer corev1informers.ServiceInformer,
+) *ImageConfigController {
+	icc := &ImageConfigController{
+		configClient: configClient,
+		routeLister: routeInformer.Lister().Routes(
+			defaults.ImageRegistryOperatorNamespace,
+		),
```

Comment:
Are you trying to fit into 80 columns?

Reply:
Sorta. In the end just put a comment where you feel like I should not indent and I will fix it for you.